### ISSUE TITLE
Get-ServiceNowTable suggestion

### ISSUE THREAD

**Issue by NFLEagles1** [Sam-Martin/servicenow-powershell#44]
Hey Sam,

Excellent Module by the way.  I just wanted to suggest you add the sysparm_fields parameter to your Get-ServiceNowTable.

I have added it to my own copy of this module.  In param:  
`        [parameter(ParameterSetName = 'SpecifyConnectionFields')]
        [parameter(ParameterSetName = 'UseConnectionObject')]
        [parameter(ParameterSetName = 'SetGlobalAuth')]
        [array]$FieldsArray,`

Then, just below "if ($Query)" statement I have added:
`    if ($FieldsArray) {
        $Body.sysparm_fields = $FieldsArray -join ","
    }`

Now, I can select the fields returned in the response.  I'll leave it to you to decide and variable names and whatnot, just wanted to add a quick suggestion.

**Comment by Rick-2CA** [comment]
Hi, @NFLEagles1 .  I just want to make sure I understand this correctly.  You want to be able to filter for only the fields you want rather than returning all of them?

**Comment by NFLEagles1** [comment]
Yes.  I do apologize for not being more clear.

**Comment by Rick-2CA** [comment]
Paging @replicaJunction - using this issue to discuss the merged PR.

Looking for opinions on how to handle the PSTypeName formatting.  If a user uses `-Fields` to specify what they want without the default properties shown by the format.ps1xml configuration then the function returns the formatted output with null values.  Two relatively easy options are available:

1. If `Fields` is not null add the default fields.  We could import the format.ps1xml file to determine what's configured to create a lookup variable.  Then would need to set the TypeName at the top so it can be referenced for the lookup and applied at the bottom.
2.  If `Fields` is not null drop formatting all together.

We could ignore this behavior as accepted too, but that strikes me as irresponsible.  Any other good ideas pop into anyone's mind?

**Comment by replicaJunction** [comment]
Good point, and I agree that just ignoring it would be irresponsible.

I'd lean towards the second option. Adding default fields, to me, feels like telling the user that what they asked for isn't really what they want. Since `Fields` can limit the data that's returned, if for some reason a user only wants to return two or three properties instead of the whole object, adding default values would cause that to return a whole bunch of extra properties instead. I'm not sure I can think of a use case for that, but I don't like limiting user choice. 😄 

There's a precedent for adding default fields, too, though. `Get-ADComputer -Properties memberOf` returns more than just the memberOf property - it still includes the properties that would be included without specifying -Properties at all.

So basically, I'd prefer to drop the formatting, but either way would work.

**Comment by Rick-2CA** [comment]
What's your thought on changing `Fields` to `Properties` as the parameter name?  Fields makes sense in the ServiceNow realm, but properties is consistent in the PowerShell realm.

Keep it fields and add a Properties alias?

**Comment by replicaJunction** [comment]
Definitely. I'd be comfortable going the other way, too - rename it to Properties and make Fields the alias.

**Comment by Rick-2CA** [comment]
Thanks NFLEagles1 for the issue and thanks replicaJunction for the effort to push this through.
